core/src/elements/mine.rs

```diff
@@ -166,9 +189,9 @@ fn update_thrown_mines(
         };
 
         let frame_time = 1.0 / crate::FPS;
-        thrown_mine.age += 1.0 / crate::FPS;
+        thrown_mine.arm_delay.tick(time.delta());
 
-        if thrown_mine.age >= *arm_delay && thrown_mine.age - *arm_delay < frame_time {
+        if thrown_mine.arm_delay.finished() && thrown_mine.arm_delay.elapsed_secs() < frame_time {
```

Comment:
`Timer` actually has a helper function that's perfect for this:

```suggestion
        if thrown_mine.arm_delay.just_finished() {
```